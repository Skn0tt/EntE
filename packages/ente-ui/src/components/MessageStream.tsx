/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import Snackbar from "@material-ui/core/Snackbar/Snackbar";
import { IconButton } from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import { MessagesConsumer } from "../context/Messages";

interface MessageStreamOwnProps {}

type MessageStreamProps = MessageStreamOwnProps;

/**
 * # Component
 */
export const MessageStream: React.SFC<MessageStreamProps> = props => {
  return (
    <MessagesConsumer>
      {({ messages, removeMessage }) =>
        messages.map((msg, i) => (
          <Snackbar
            key={i}
            message={<span>{msg}</span>}
            autoHideDuration={6000}
            onClose={(_, reason) =>
              reason !== "clickaway" && removeMessage(msg)
            }
            open
            action={
              <IconButton
                onClick={() => removeMessage(msg)}
                color="inherit"
                className={"remove" + i}
              >
                <CloseIcon />
              </IconButton>
            }
          />
        ))
      }
    </MessagesConsumer>
  );
};

export default MessageStream;
