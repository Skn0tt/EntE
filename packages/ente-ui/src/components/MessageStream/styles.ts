/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { Theme, StyleRules } from "material-ui/styles";

const styles = (theme: Theme): StyleRules => ({});

export default styles;
