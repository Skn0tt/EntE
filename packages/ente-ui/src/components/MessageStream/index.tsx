/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import styles from "./styles";
import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
import { Dispatch, Action } from "redux";
import { connect } from "react-redux";
import Snackbar from "@material-ui/core/Snackbar/Snackbar";
import { IconButton } from "@material-ui/core";
import { Close as CloseIcon } from "@material-ui/icons";
import { AppState, removeMessage, getMessages } from "ente-redux";

/**
 * # Component Types
 */
interface StateProps {
  messages: String[];
}
const mapStateToProps = (state: AppState) => ({
  messages: getMessages(state)
});

interface DispatchProps {
  removeMessage(index: number): Action;
}
const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  removeMessage: (index: number) => dispatch(removeMessage(index))
});

type Props = StateProps & DispatchProps & WithStyles;

/**
 * # Component
 */
export const MessageStream: React.SFC<Props> = props => {
  const { messages, removeMessage } = props;

  return (
    <>
      {messages.map((msg, i) => (
        <Snackbar
          key={i}
          message={<span>{msg}</span>}
          autoHideDuration={6000}
          onClose={(_, reason) => reason !== "clickaway" && removeMessage(i)}
          open
          action={
            <IconButton
              onClick={() => removeMessage(i)}
              color="inherit"
              className={"remove" + i}
            >
              <CloseIcon />
            </IconButton>
          }
        />
      ))}
    </>
  );
};

export default withStyles(styles)(
  connect(mapStateToProps, mapDispatchToProps)(MessageStream)
);
