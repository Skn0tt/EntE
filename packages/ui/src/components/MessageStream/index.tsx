import * as React from "react";
import styles from "./styles";
import withStyles, { WithStyles } from "material-ui/styles/withStyles";
import { Dispatch, Action } from "redux";
import { connect } from "react-redux";
import Snackbar from "material-ui/Snackbar/Snackbar";
import { IconButton } from "material-ui";
import { Close as CloseIcon } from "material-ui-icons";
import { AppState, removeMessage, getMessages } from "ente-redux";

interface StateProps {
  messages: String[];
}
const mapStateToProps = (state: AppState) => ({
  messages: getMessages(state)
});

interface DispatchProps {
  removeMessage(index: number): Action;
}
const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  removeMessage: (index: number) => dispatch(removeMessage(index))
});

type Props = StateProps & DispatchProps & WithStyles;

const MessageStream: React.SFC<Props> = props => (
  <React.Fragment>
    {props.messages.map((msg, index) => (
      <Snackbar
        key={index}
        message={<span>{msg}</span>}
        autoHideDuration={6000}
        onClose={(event, reason) =>
          reason !== "clickaway" && props.removeMessage(index)
        }
        open
        action={
          <IconButton
            onClick={() => props.removeMessage(index)}
            color="inherit"
          >
            <CloseIcon />
          </IconButton>
        }
      />
    ))}
  </React.Fragment>
);

export default connect(mapStateToProps, mapDispatchToProps)(
  withStyles(styles)(MessageStream)
);
