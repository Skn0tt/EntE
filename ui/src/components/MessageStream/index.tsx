import * as React from 'react';
import styles from './styles';
import withStyles, { WithStyles } from 'material-ui/styles/withStyles';
import { AppState } from '../../interfaces/index';
import * as select from '../../redux/selectors';
import { Dispatch, Action } from 'redux';
import { connect } from 'react-redux';
import Snackbar from 'material-ui/Snackbar/Snackbar';
import { IconButton } from 'material-ui';
import { Close as CloseIcon } from 'material-ui-icons';
import { removeMessage } from '../../redux/actions';

interface StateProps {
  messages: String[];
}

interface DispatchProps {
  removeMessage(index: number): Action;
}

type Props = StateProps & DispatchProps & WithStyles;

const MessageStream: React.SFC<Props> = props => (
  <div>
    {props.messages.map((msg, index) => (
      <Snackbar
        key={index}
        message={<span>{msg}</span>}
        autoHideDuration={6000}
        onClose={() => props.removeMessage(index)}
        open
        action={
          <IconButton
            onClick={() => props.removeMessage(index)}
            color="inherit"
          >
            <CloseIcon />
          </IconButton>
        }
      />
    ))}
  </div>
);

const mapStateToProps = (state: AppState) => ({
  messages: select.getMessages(state),
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  removeMessage: (index: number) => dispatch(removeMessage(index)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(MessageStream));
