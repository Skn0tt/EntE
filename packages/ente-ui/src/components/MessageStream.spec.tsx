/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import { shallow } from "enzyme";
import { MessageStream } from "./MessageStream";

describe("MessageStream", () => {
  const removeMessage = jest.fn();
  const messages = ["Hallo", "Nachricht"];

  const comp = shallow(
    <MessageStream
      messages={messages}
      removeMessage={removeMessage}
      classes={{}}
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });
});
