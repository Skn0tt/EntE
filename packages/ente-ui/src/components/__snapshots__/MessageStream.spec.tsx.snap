// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageStream renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessagesProvider>
    <Unknown />
  </MessagesProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown />,
      "value": Object {
        "addMessages": [Function],
        "messages": Array [],
        "removeMessage": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "addMessages": [Function],
          "messages": Array [],
          "removeMessage": [Function],
        },
        "_currentValue2": Object {
          "addMessages": [Function],
          "messages": Array [],
          "removeMessage": [Function],
        },
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown />,
        "value": Object {
          "addMessages": [Function],
          "messages": Array [],
          "removeMessage": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "addMessages": [Function],
            "messages": Array [],
            "removeMessage": [Function],
          },
          "_currentValue2": Object {
            "addMessages": [Function],
            "messages": Array [],
            "removeMessage": [Function],
          },
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
